src/Domains/Registrar/Mock.php

```
@@ -0,0 +1,402 @@
+    }
+
+    /**
+     * Purchase a domain
+     *
+     * @param string $domain
+     * @param array<\Utopia\Domains\Contact> $contacts
+     * @param array $nameservers
+     * @return array
+     * @throws DomainTaken
+     */
+    public function purchase(string $domain, array|Contact $contacts, array $nameservers = []): array
```

PHPMD (2.15.0)

Avoid unused parameters such as '$contacts'. (undefined)

(UnusedFormalParameter)